### PR TITLE
Pact Layer

### PR DESCRIPTION
**Pull request by fosskers**
### TODO
- [x] Initial detection of `pact-mode`
- [x] Integration w/ `pact` repl
- [x] `flycheck` integration
- [x] README

### Motivation
This PR adds a language layer for the Smart Contract language [Pact](https://github.com/kadena-io/pact).

### CODE REVIEW

File: layers/+lang/pact/packages.el [hunk 1]
```diff
@@ -0,0 +1,50 @@
+;;; packages.el --- Pact Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2018 Kadena LLC
```
Comment (sdwolfz) [inline layers/+lang/pact/packages.el]: Note, this line should be:
```emacs-lisp
;; Copyright (c) 2012-2018 Sylvain Benner & Contributors
```

Comment (fosskers) [reply layers/+lang/pact/packages.el]: Ah, my mistake, I had copy-pasted that originally from the `racket layer and assumed those names were the authors of that layer, not of Spacemacs itself. I'll correct it.

**Comment by robbyoconnor** [comment]
Fix this: https://circleci.com/gh/syl20bnr/spacemacs/14860?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by fosskers** [comment]
@robbyoconnor just did, thanks.

**Comment by fosskers** [comment]
I can't seem to get `flycheck` to work. I've referenced other layer that use it, but when I open a `.pact` file I don't see it turn nor (and thus no red squigglies, etc.).

**Comment by fosskers** [comment]
@robbyoconnor It doesn't look like `SPC e s` is bound to anything. Did you mean something else?

**Comment by fosskers** [comment]
For the record, there is a [flycheck-pact](https://github.com/kadena-io/flycheck-pact) package, but as I'm not the original author I'm not sure if it's written in a way that Spacemacs would expect.

**Comment by fosskers** [comment]
Gotta rebase again. One moment...

**Comment by fosskers** [comment]
Hm, even adding `(add-hook 'pact-mode-hook 'flycheck-mode)` to my `dotspacemacs/user-config` (as I need to for both Purescript and Emacs Lisp, apparently) doesn't work.

**Comment by fosskers** [comment]
Here is a suitable file to test on, for anyone interested:
```
(cash.create-account "stu" "USD" 1000.0 (time "2016-09-01T11:00:00Z"))
```

**Comment by fosskers** [comment]
I consistently get the following in my modeline upon opening a pact file:
```
Error during redisplay: (jit-lock-function 1629) signaled (end-of-buffer)
```

**Comment by fosskers** [comment]
Here's a good test file: https://github.com/kadena-io/pact/blob/master/examples/accounts/accounts.pact

**Comment by fosskers** [comment]
CI is failing for reasons unrelated to this PR:
```
...
Checking /root/.emacs.d/elpa/25.2/develop/nameless-1.0.2...
Compiling /root/.emacs.d/elpa/25.2/develop/nameless-1.0.2/nameless-autoloads.el...
Compiling /root/.emacs.d/elpa/25.2/develop/nameless-1.0.2/nameless-pkg.el...
Compiling /root/.emacs.d/elpa/25.2/develop/nameless-1.0.2/nameless.el...
Done (Total of 1 file compiled, 2 skipped)
(Spacemacs) --> installing package: overseer@emacs-lisp... [15/15]
Cannot open load file: No such file or directory, bind-map
../../../../spacemacs.mk:20: recipe for target 'installation' failed
make: *** [installation] Error 255
Exited with code 2
```

**Comment by sdwolfz** [comment]
Thank you :heart:!
Cherry-picked into develop branch, you can safely delete your branch.

**Comment by fosskers** [comment]
Thank you!

**Pull request merged**